include/seqan3/search/fm_index/fm_index.hpp

```diff
@@ -166,6 +174,9 @@ class fm_index
     using iterator_type = fm_index_iterator<fm_index<text_t, fm_index_traits>>;
     //!\}
 
+    //!\brief Indicates whether index is built over a collection
+    static bool constexpr is_collection = dimension_v<text_t> == 2;
```

Comment:
what is with dimension 3 and higher?

Reply:
Well, I wouldn't know what to do with it.
If it's an collection of collections, it would need to be flattened to a collection to build an index on it.
I'd guess currently it will crash during compiling, because something doesn't work - just like if you try to index a collection in the current version.

Another question would be, if this could be treated more elegantly, i.e. you actually know that it fails because you have to many dimensions.
Maybe adjust the `requires` to only be defined for dimension==1 and dimension==2?

Reply:
you can make a static_assert that the dimension is 1 or 2